spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binder/AbstractMessageChannelBinder.java

```diff
@@ -366,4 +522,62 @@ public boolean isRunning() {
 			return this.delegate instanceof Lifecycle && ((Lifecycle) this.delegate).isRunning();
 		}
 	}
+
+	private interface SubscriberAwareChannel {
```

Comment:
Looks like nobody implements this interface.
I guess the idea was to make that `ErrorChannel` below to follow this contract.

Although looking to that `FinalRethrowingErrorMessageHandler` it really doesn't matter if use has subscribers or not.
You claim there like `run last if the user has subscribers too`.

Nevertheless maybe that would be much better to **remove** that `FinalRethrowingErrorMessageHandler` from the `ErrorChannel` just in the moment use calls its `subscribe()`.